modules/autoattach.cpp

```diff
@@ -151,38 +167,39 @@ class CChanAttach : public CModule {
                    [=](const CString& sLine) { HandleDel(sLine); });
         AddCommand("List", "", t_d("List all entries"),
                    [=](const CString& sLine) { HandleList(sLine); });
+        AddCommand("Swap", t_d("<a> <b>"), t_d("Swap a and b entries in the list."),
+                   [=](const CString& sLine) { HandleSwap(sLine); });
     }
 
     ~CChanAttach() override {}
 
     bool OnLoad(const CString& sArgs, CString& sMessage) override {
+        CString migrated = GetNV("migrated");
+
+        if (migrated != "1") {
+            MigrateFromOldNV();
+        }
+
         VCString vsChans;
-        sArgs.Split(" ", vsChans, false);
+
+        GetNV("Autoattach").Split("\n", vsChans, false);
+
+        for (const CString& sRule : vsChans) {
+            if (!AddFromString(sRule)) {
+                PutModule(t_f("Unable to add [{1}]")(sRule));
+            }
+        }
 
         for (VCString::const_iterator it = vsChans.begin(); it != vsChans.end();
              ++it) {
             CString sAdd = *it;
-            bool bNegated = sAdd.TrimPrefix("!");
-            CString sChan = sAdd.Token(0);
-            CString sSearch = sAdd.Token(1);
-            CString sHost = sAdd.Token(2, true);
 
-            if (!Add(bNegated, sChan, sSearch, sHost)) {
+            if (!AddFromString(sAdd)) {
                 PutModule(t_f("Unable to add [{1}]")(*it));
             }
         }
 
-        // Load our saved settings, ignore errors
-        MCString::iterator it;
-        for (it = BeginNV(); it != EndNV(); ++it) {
-            CString sAdd = it->first;
-            bool bNegated = sAdd.TrimPrefix("!");
-            CString sChan = sAdd.Token(0);
-            CString sSearch = sAdd.Token(1);
-            CString sHost = sAdd.Token(2, true);
-
-            Add(bNegated, sChan, sSearch, sHost);
-        }
+        Save();
```

Comment:
Why?

Reply:
Because I apparently can't read and misunderstood was some of what the original code was doing. Push coming soon.